### DISCUSSION TITLE
无法启动REST服务

### DISCUSSION THREAD

**Discussion by jason-mua** [apache/iotdb#5620] [General]
![image](https://user-images.githubusercontent.com/32285735/164361951-f9d96add-dbc8-4c1b-8265-c70933114b39.png)
在执行sbin/start-server.sh的时候，发现REST无法启动。18080端口没有被占用。在另外一台机器上也安装了IoTDB，可以启动REST服务。下载的源码包缺少很多依赖，无法在idea里面调试，我本身也不了解jetty，大概是什么原因导致的REST无法启动的？

**Comment by jason-mua** [comment]
![image](https://user-images.githubusercontent.com/32285735/164416187-8e2912fa-2cd7-4b18-a123-a3f13af05a01.png)
发现lib目录下放置了hadoop-tsfile-0.13.0-jar-with-dependencies.jar这个jar包，就无法启动RestService了，删除了这个包里的javax目录就可以了。有知道具体原因的大佬，可以解答一下

**Comment by jixuan1989** [comment]
we can find the reason by using debug level log:
```
Suppressed: java.lang.NoSuchMethodError: javax.servlet.ServletContext.createFilter(Ljava/lang/Class;)Ljavax/servlet/Filter;
```

Such kind of errors are  mainly caused by the conflict of dependencies. 

By checking the dependencies using `mvn dependency:tree`, we can find Hadoop depends on javax.servlet 2.5, and iotdb depends on 3.1  (introduced by org.eclipse.jetty:jetty-server:jar:9.4.35.v20201120 in the server module).

Need contributors to check whether can we downgrade to  javax.servlet 2.5 in the server module, or upgrade the hadoop module dependency.

**Reply by jixuan1989** [reply]
I have create an issue about this. 
https://issues.apache.org/jira/browse/IOTDB-3051
this is an easy-fix bug and welcome to contribute on it.
